internal/metrics/metrics.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const defaultInterval = 10 * time.Second
+
+var logger log.LeveledLogger = log.NewFromGlobal(log.AddContext("pkg", "metrics"))
+
+// Config is base config for metrics
+type Config struct {
+	Publish bool
+}
+
+// IntervalConfig for interval collection
+type IntervalConfig struct {
+	Config
+	Interval time.Duration
+}
+
+// NewIntervalConfig is constructor for IntervalConfig, and uses default metrics interval
+func NewIntervalConfig(publish bool) IntervalConfig {
+	return IntervalConfig{
+		Config:   Config{publish},
+		Interval: defaultInterval,
+	}
+}
+
+// Start will start a dedicated metrics server at the given address.
+func Start(address string) {
+	logger.Info("Starting metrics server at http://" + address + "/metrics")
+	http.Handle("/metrics", promhttp.Handler())
+	go func() {
+		if err := http.ListenAndServe(address, nil); err != nil {
+			log.Errorf("Metrics HTTP server crashed: %s", err)
+		}
+	}()
+}
```

Comment:
You could use `internal/httpserver` which has semantic for readiness and controlled ending.

I would vouch against go routine you 'fire and forget' like this one. And also global function are not great, it's better to have a constructor with a start method on it.

You can use the [pprof server](https://github.com/ChainSafe/gossamer/blob/e4bc3757c2011a387d34baba64c8a5bb9aee1a81/internal/pprof/server.go#L27) constructor as an example:

```go
func NewServer(address string, logger httpserver.Logger) *httpserver.Server {
	handler := http.NewServeMux()
	handler.HandleFunc("/debug/pprof/", pprof.Index)
	// ...
	return httpserver.New("pprof", address, handler, logger, options...)
}
```

To use it as a service, you can use something like [this](https://github.com/ChainSafe/gossamer/blob/e4bc3757c2011a387d34baba64c8a5bb9aee1a81/internal/pprof/service.go#L25)

Reply:
I agree. I pretty much copy pasta'd this code from `dot/metrics`.